include/fastdds/rtps/common/LocatorSelectorEntry.hpp

```diff
@@ -100,6 +101,26 @@ struct LocatorSelectorEntry
         state.multicast.clear();
     }
 
+    void fill_unicast(
+            const LocatorList_t& locators)
+    {
+        for (const Locator_t& locator : locators)
+        {
+            state.unicast.push_back(unicast.size());
+            unicast.push_back(locator);
+        }
+    }
+
+    void fill_multicast(
+            const LocatorList_t& locators)
+    {
+        for (const Locator_t& locator : locators)
+        {
+            state.multicast.push_back(multicast.size());
+            multicast.push_back(locator);
+        }
+    }
+
```

Comment:
Given the usage we are going to give to these methods, we could use a builder pattern instead.

```suggestion
    static LocatorSelectorEntry create_fully_selected_entry(
            const LocatorList_t& unicast_locators,
            const LocatorList_t& multicast_locators)
    {
        // Create an entry with space for all locators
        LocatorSelectorEntry entry(unicast_locators.size(), multicast_locators.size());

        // Add and select unicast locators
        for (const Locator_t& locator : unicast_locators)
        {
            entry.state.unicast.push_back(unicast.size());
            entry.unicast.push_back(locator);
        }

        // Add and select multicast locators
        for (const Locator_t& locator : multicast_locators)
        {
            entry.state.multicast.push_back(multicast.size());
            entry.multicast.push_back(locator);
        }

        // Return created entry
        return entry;
    }

```